internal/controller/server_controller.go

```diff
@@ -665,16 +665,37 @@ func (r *ServerReconciler) ensureServerPowerState(ctx context.Context, log logr.
 		if err := bmcClient.PowerOn(server.Spec.UUID); err != nil {
 			return fmt.Errorf("failed to power on server: %w", err)
 		}
+		if err := waitForServerPowerState(ctx, log, bmcClient, server, redfish.OnPowerState); err != nil {
+			return fmt.Errorf("failed to wait for server power on server: %w", err)
+		}
 	case powerOpOff:
 		if err := bmcClient.PowerOff(server.Spec.UUID); err != nil {
 			return fmt.Errorf("failed to power off server: %w", err)
 		}
+		if err := waitForServerPowerState(ctx, log, bmcClient, server, redfish.OffPowerState); err != nil {
+			return fmt.Errorf("failed to wait for server power off server: %w", err)
+		}
 	}
 	log.V(1).Info("Ensured server power state", "PowerState", server.Spec.Power)
 
 	return nil
 }
 
+func waitForServerPowerState(ctx context.Context, log logr.Logger, bmcClient bmc.BMC, server *metalv1alpha1.Server, powerState redfish.PowerState) error {
+	if err := wait.PollUntilContextTimeout(ctx, 1*time.Second, 10*time.Second, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
should we bump the timeout to allow for baremetal to properly power on/off? Maybe 2 minutes?

Reply:
Addressed this via https://github.com/ironcore-dev/metal-operator/pull/128/commits/f35ecb925b0b8d7f89bc9fe1124ce49b60d1fda4